src/Datadog.Trace/Configuration/JsonConfigurationSource.cs

```diff
@@ -97,5 +99,40 @@ public T GetValue<T>(string key)
                        ? default
                        : token.Value<T>();
         }
+
+        /// <summary>
+        /// Gets a <see cref="ConcurrentDictionary{TKey, TValue}"/> containing all of the values.
+        /// </summary>
+        /// <remarks>
+        /// Example JSON where `globalTags` is the configuration key.
+        /// {
+        ///  "globalTags": {
+        ///     "name1": "value1",
+        ///     "name2": "value2"
+        ///     }
+        /// }
+        /// </remarks>
+        /// <param name="key">The key that identifies the setting.</param>
+        /// <returns><see cref="IDictionary{TKey, TValue}"/> containing all of the key-value pairs.</returns>
+        /// <exception cref="JsonReaderException">Thrown if the configuration value is not a valid JSON string.</exception>"
+        public IDictionary<string, string> GetDictionary(string key)
+        {
+            var token = _configuration.SelectToken(key, errorWhenNoMatch: false);
+            if (token == null)
+            {
+                return null;
+            }
+
+            // Presence of a curly brace indicates that this is likely a JSON string. An exception will be thrown
+            // if it fails to parse or convert to a dictionary.
+            if (token.ToString().Contains("{"))
+            {
+                var dictionary = JToken.Parse(token.ToString())
```

Comment:
Similar to above, `token.ToString()` serializes the token back into a json string, only to deserialize it a second time with `JToken.Parse()`. We should be able drop these two calls and use `token.ToObject(...)` directly.
```suggestion
                var dictionary = token
```

Reply:
👍